backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/impexp/CampaignPriceImportTableSqlUpdater.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.pricing.rules.campaign_price.impexp;
+
+import de.metas.impexp.processing.ImportRecordsSelection;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.experimental.UtilityClass;
+import org.adempiere.ad.trx.api.ITrx;
+import org.compiere.model.I_I_Campaign_Price;
+import org.compiere.util.DB;
+
+import java.util.Properties;
+
+import static de.metas.impexp.format.ImportTableDescriptor.COLUMNNAME_I_ErrorMsg;
+import static de.metas.impexp.format.ImportTableDescriptor.COLUMNNAME_I_IsImported;
+
+@UtilityClass
+public class CampaignPriceImportTableSqlUpdater
+{
+	private final static String targetTableName = I_I_Campaign_Price.Table_Name;
+
+	@Builder(buildMethodName = "updateCampaignPrice")
+	private void updateCampaignPrice(@NonNull final ImportRecordsSelection selection, @NonNull final Properties ctx)
+	{
+		dbUpdateProduct(selection);
+		dbUpdateBPartner(selection);
+		dbUpdateBpGroup(selection);
+		dbUpdatePricingSystem(selection);
+		dbUpdateCountry(selection);
+		dbUpdateCurrency(selection);
+		dbUpdateUom(selection);
+		dbUpdateTaxCategory(selection);
+		dbUpdateCampaignPrice(selection, ctx);
+	}
+
+	private static void dbUpdateProduct(final ImportRecordsSelection selection)
+	{
+		StringBuilder sql = new StringBuilder("UPDATE ")
+				.append(targetTableName + " i ")
+				.append(" SET M_Product_ID=(SELECT M_Product_ID FROM M_Product p")
+				.append(" WHERE i.ProductValue=p.Value AND p.AD_Client_ID in (i.AD_Client_ID,0) AND p.AD_Org_ID IN (0, i.AD_Org_ID) LIMIT 1) ")
```

Comment:
Makes sense, I did something similar but would up ditching it because it  seemed overkill on client's DB. Changed
![image](https://user-images.githubusercontent.com/60300682/165274560-d6f04a8a-5361-4b42-bd08-3eef1dd51c7a.png)